src/shared/tests/pipes.test.tsx

```diff
@@ -0,0 +1,247 @@
+    expect(endDatePipe(new Date('12/25/20'), 3)).toBe('1/15/2021');
+    expect(endDatePipe(new Date('1/3/21'), 3)).toBe('1/24/2021');
+    expect(endDatePipe(new Date('3/1/21'), 10)).toBe('5/10/2021');
+  });
+});
+
+describe('Formatting Links Tests', () => {
+  test('with known websites', () => {
+    expect(linkPipe('google', 'www.google.com')).toStrictEqual(
+      <a href={'www.google.com'}>{'google'}</a>
+    );
+    expect(linkPipe('facebook', 'www.facebook.com')).toStrictEqual(
+      <a href={'www.facebook.com'}>{'facebook'}</a>
+    );
+    expect(linkPipe('github', 'www.github.com')).toStrictEqual(
+      <a href={'www.github.com'}>{'github'}</a>
+    );
+  });
+});
+
+describe('Formatting Weeks Tests', () => {
+  test('with 1', () => {
+    expect(weeksPipe(1)).toBe('1 week');
+  });
+  test('with int > 1', () => {
+    expect(weeksPipe(2)).toBe('2 weeks');
+  });
+  test('with 0', () => {
+    expect(weeksPipe(0)).toBe('0 weeks');
+  });
+  test('with decimal', () => {
+    expect(weeksPipe(1.5)).toBe('1.5 weeks');
+  });
+  test('with hexadecimal', () => {
+    expect(weeksPipe(0x37cf)).toBe('14287 weeks');
+  });
+  test('with octal', () => {
+    expect(weeksPipe(0o377)).toBe('255 weeks');
+  });
+  test('with binary', () => {
+    expect(weeksPipe(0b111001)).toBe('57 weeks');
+  });
+});
+
+describe('Formatting Dollars Tests', () => {
+  test('with 1', () => {
+    expect(dollarsPipe(1)).toBe('$1');
+  });
+  test('with int > 1', () => {
+    expect(dollarsPipe(2)).toBe('$2');
+  });
+  test('with 0', () => {
+    expect(dollarsPipe(0)).toBe('$0');
+  });
+  test('with decimal', () => {
+    expect(dollarsPipe(6.9)).toBe('$6.9');
+    expect(dollarsPipe(4.2)).toBe('$4.2');
+  });
+  test('with hexadecimal', () => {
+    expect(dollarsPipe(0x37cf)).toBe('$14287');
+  });
+  test('with octal', () => {
+    expect(dollarsPipe(0o377)).toBe('$255');
+  });
+  test('with binary', () => {
+    expect(dollarsPipe(0b111001)).toBe('$57');
+  });
+});
+
+describe('Formatting Booleans', () => {
+  test('Formatting booleans', () => {
+    expect(booleanPipe(true)).toBe('YES');
+    expect(booleanPipe(false)).toBe('NO');
+  });
+});
+
+describe('Formatting Empty Strings Without Dash', () => {
+  test('empty string', () => {
+    expect(emptyStringPipe('')).toBe('');
+  });
+  test('empty string with space', () => {
+    expect(emptyStringPipe(' ')).toBe(' ');
+  });
+  test('non-empty string', () => {
+    expect(emptyStringPipe('abcd')).toBe('abcd');
+  });
+  test('non-empty string with numbers', () => {
+    expect(emptyStringPipe('123')).toBe('123');
+  });
+  test('non-empty string with special characters', () => {
+    expect(emptyStringPipe('!@#$%^&*()}{:">?')).toBe('!@#$%^&*()}{:">?');
+  });
+});
+
+describe('Formatting Empty Strings With Dash', () => {
+  test('empty string', () => {
+    expect(emDashPipe('')).toBe('');
+  });
+  test('empty string with space', () => {
+    expect(emDashPipe(' ')).toBe(' ');
```

Comment:
I think it'd be good to have this behave by outputting an em dash. Check out the trim method if you're not already familiar with it